### PR TITLE
Added support for periodic GC on config.git repo

### PR DESCRIPTION
**Pull request by jyotisingh**
* jgit upgraded to 4.1
* periodic GC that triggers at 1am, can be overridden using 
``` properties
-Dgo.config.gc.cron=<cron expression>
```
* GC happens in aggressive mode by default. Can be turned off using 
``` properties
-Dgo.config.repo.gc.aggressive=false
```
* using SLF4j logger instead

### CODE REVIEW

File: config/config-server/src/com/thoughtworks/go/service/ConfigRepository.java [hunk 1]
```diff
@@ -390,8 +390,23 @@ void cleanAndResetToMaster() throws IOException {
...
         }
     }
+
+    public void garbageCollect() throws Exception {
+        doLocked(new VoidThrowingFn<Exception>() {
+            public void run() throws Exception {
+                try {
+                    LOGGER.info("Before GC: {}", git.gc().getStatistics());
+                    git.gc().setAggressive(systemEnvironment.get(SystemEnvironment.GO_CONFIG_REPO_GC_AGGRESSIVE)).call();
+                    LOGGER.info("After GC: {}", git.gc().getStatistics());
```
Comment (ketan) [inline config/config-server/src/com/thoughtworks/go/service/ConfigRepository.java]: `Before GC` and `After GC` could be confusing (heap GC?). Can we clarify this a bit?

Comment (jyotisingh) [inline config/config-server/src/com/thoughtworks/go/service/ConfigRepository.java]: The log file entry would look something like this:
```
2015-11-05 19:31:00,029  INFO [pool-1-thread-1] ConfigRepository:403 - Before running GC: {sizeOfPackedObjects=0, sizeOfLooseObjects=6640456, numberOfPackedObjects=0, numberOfPackFiles=0, numberOfPackedRefs=0, numberOfLooseRefs=2, numberOfLooseObjects=2037}
2015-11-05 19:31:02,199  INFO [pool-1-thread-1] ConfigRepository:405 - After running GC: {sizeOfPackedObjects=415262, sizeOfLooseObjects=0, numberOfPackedObjects=2037, numberOfPackFiles=2, numberOfPackedRefs=1, numberOfLooseRefs=1, numberOfLooseObjects=0}
```
That calls out the class name `ConfigRepository`, so I guess that should be ok.

**Comment by ketan** [comment]
@jyotisingh on second thought can the properties be —

````
-Dgo.config.repo.gc.cron=<cron expression>
-Dgo.config.repo.gc.aggressive=false
```

**Comment by lenucksi** [comment]
That probably also tackles #1567.

**Comment by jyotisingh** [comment]
@ketan - #1601

**Pull request merged**